lstchain/scripts/lstchain_find_pedestals.py

```diff
@@ -0,0 +1,173 @@
+                           'pedestal_ids_Run{:0>5}.{:0>4}.h5'.format(run_number,
+                                                                     subrun_index))
+        df.to_hdf(output_file, key='interleaved_pedestal_ids', mode='w')
+
+
+def find_pedestals(timestamps, aprox_frequency=50):
+    """
+    Parameters
+    ----------
+    timestamps: ndarray, series of timestamps of LST1 events from a sub-run
+    aprox_frequency: approximate frequency (Hz) of the interleaved pedestals
+    for the epoch of the observations
+
+    Returns
+    a mask which is True for the elements of timestamps which are most evenly
+    distributed (and with the given approximate frequency)
+
+    The function finds the set of events which are most evenly distributed
+    in time with the approximate frequency aprox_frequency. This should be the
+    interleaved pedestals, which are very regularly spaced in time to better
+    (typically) than microsec precision. Better to pass an array of
+    timestamps in which flatfield events (easily recognizable) have been
+    excluded.
+
+    -------
+
+    """
+    period_mean = 1 / aprox_frequency
+    period_step_width = 1e-7  # s
+    period_step_range = np.linspace(-50, 50, 101)
+
+    phase = None
+    best_contents = None
+    best_bins = None
+    best_period = None
+
+    average_nevents_per_bin = 1
+    # number of bins for the phase-folded histogram:
+    nbins = int(len(timestamps) / average_nevents_per_bin)
+    # aprox average_nevents_per_bin cosmics will remain per bin
+    # (i.e. a contamination for the genuine pedestals we are looking for)
+
+    # Find the period:
+    max_peak = 0
+    for i in period_step_range:
+        tmod = timestamps % (period_mean + i * period_step_width)
+        contents, bins = np.histogram(tmod, bins=nbins, range=(0, period_mean))
+        if np.max(contents) < max_peak:
+            continue
+        max_peak = np.max(contents)
+        best_bins = bins
+        best_contents = contents
+        best_period = period_mean + i * period_step_width
+    # print('best_period =', best_period, ", best freq = ", 1/best_period)
+
+    # adjust phase:
+    max_peak = 0
+    best_phase = 0
+    steps = 1000  # full phase will be divided in this number of steps
+
+    for phase in np.linspace(0, best_period, steps):
+        tmod = (timestamps + phase) % best_period
+        contents, bins = np.histogram(tmod, bins=nbins, range=(0, best_period))
+        if np.max(contents) < max_peak:
+            continue
+        max_peak = np.max(contents)
+        best_phase = phase
+        best_contents = contents
+        best_bins = bins
+
+    tmod = (timestamps + best_phase) % best_period
+
+    ibin = np.argmax(best_contents)
+    # check the two neighboring bins, which may also contain pedestals.
+    # Include them if content is at least 10% of the max:
+    ifirst = ibin
+    ilast = ibin
+    if best_contents[ibin - 1] > 0.1 * best_contents[ibin]:
```

Comment:
This will error if bin 0 or bin N is the largest bin. I think you want circular boundary conditions for this, right? 

so 

```
bin_before = (best_bin + len(best_contents) - 1) % len(best_contents)
bin_after = (best_bin + 1) % len(best_contents)
```

Reply:
yes, thanks, will do!